client/src/main/java/io/pravega/client/batch/impl/SegmentIteratorImpl.java

```diff
@@ -95,4 +103,65 @@ public void close() {
         input.close();
     }
 
-}
+    public static class SegmentIteratorPosition {
```

Comment:
I may have lead you astray.
There already is a class with these same fields `SegmentRange`. Perhaps we should use it. 
It may be useful for it to be serializable. Perhaps via toString and fromString methods similar to `Segment`

Reply:
No problem, we can make use Segment Range.